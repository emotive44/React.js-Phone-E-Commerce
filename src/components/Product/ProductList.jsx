import React, { Fragment } from 'react';
import Product from './Product';
import Title from './Title';
import { ProductConsumer } from '../../context'

class ProductList extends React.Component {
    render() {
        return(
            <Fragment>
                <div className="py-5">
                    <div className="container">
                        <Title name="our" title="products"/>
                        <div className="row">
                            <ProductConsumer>
                                {(data) => {
                                    return data.products.map( product => {
                                        return <Product key={product.id} {...product} />
                                    });
                                }}
                            </ProductConsumer>
                        </div>
                    </div>
                </div>
            </Fragment>
        )
    }
}

export default ProductList;