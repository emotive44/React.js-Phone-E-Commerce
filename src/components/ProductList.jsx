import React from 'react';
import Product from './Product';

class ProductList extends React.Component {
    render() {
        return(
            <div>
                ProductList
            </div>
        )
    }
}

export default ProductList;