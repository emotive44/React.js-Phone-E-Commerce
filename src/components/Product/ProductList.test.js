import React from 'react';
import ProductList from './ProductList';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom', () => ({
    Link: 'Link'
}));

describe('ProductList Component', () => {
    test('should render return null if products is undefiend', () => {
        const component = renderer.create(<ProductList />);
        expect(component.toJSON()).toMatchSnapshot();
    })
})