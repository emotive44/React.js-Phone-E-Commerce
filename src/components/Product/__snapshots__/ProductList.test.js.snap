// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList Component should render return null if products is undefiend 1`] = `
<div
  className="py-5"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-10 mx-auto my-2 text-center"
      >
        <h1
          className="text-capitalize font-weight-bold"
        >
          our
        </h1>
        <h3
          className="text-capitalize font-weight-bold"
        >
          products
        </h3>
      </div>
    </div>
    <div
      className="row"
    />
  </div>
</div>
`;
